src/core/hle/libraries/libkernel/file_system.cpp

```diff
@@ -9,10 +15,83 @@ constexpr bool log_file_fs = true;  // disable it to disable logging
 
 int PS4_SYSV_ABI sceKernelOpen(const char* path, int flags, u16 mode) {
     LOG_INFO_IF(log_file_fs, "sceKernelOpen path = {} flags = {:#x} mode = {:#x}\n", path, flags, mode);
-    return 0;
+    bool isDirectory = (flags & SCE_KERNEL_O_DIRECTORY) != 0;
+    bool create = (flags & SCE_KERNEL_O_CREAT) != 0;
+
+    auto* h = Common::Singleton<Core::FileSys::HandleTable>::Instance();
+    auto* mnt = Common::Singleton<Core::FileSys::MntPoints>::Instance();
+
+    u32 handle = h->createHandle();
+    if (handle >= 2048) {  // max descriptor reached
+        return SCE_KERNEL_ERROR_EMFILE;
+    }
+    auto* file = h->getFile(handle);
+    if (isDirectory) {
```

Comment:
Can be changed to early return, to reduce indentation and make it more readable

Reply:
hmm not sure what you mean...